### PR TITLE
Added cNFT Transfer

### PR DESCRIPTION
**Pull request by AshenI99**
Added token transfer for `cNFT`.

### CODE REVIEW

File: as/cNFT/assembly/models/persistent_tokens.ts [hunk 1]
```diff
@@ -113,6 +113,50 @@ export class PersistentTokens {
         return token
     }
 
+    /**
+     * @param tokenId: Id of the token
+     * @param bidderId: accountId of the bidder
+    */
+
+    cnft_transfer(tokenId: TokenId, bidderId: AccountId): void {
```
Comment (achrafismyname) [inline as/cNFT/assembly/models/persistent_tokens.ts:121]: You can call this function just `transfer`

File: as/cNFT/assembly/models/persistent_tokens.ts [hunk 2]
```diff
@@ -113,6 +113,50 @@ export class PersistentTokens {
...
+
+        /* Getting stored token from tokenId */
+
+        const token = this._tmap.getSome(tokenId)
```
Comment (achrafismyname) [inline as/cNFT/assembly/models/persistent_tokens.ts:125]: I think it's better to get the token using `this.get(tokenId)`. So in future, we have to change it in just one place

File: as/cNFT/assembly/models/persistent_tokens.ts [hunk 3]
```diff
@@ -113,6 +113,50 @@ export class PersistentTokens {
...
+
+        this._tmap.set(tokenId, token)
+
+
+        /* Deleting token from previous owner */
+
+        this._deleteFromAccountTokenSet(tokenId, accountId);
+
+
+        /* Storing token with the new owner's accountId */
```
Comment (achrafismyname) [inline as/cNFT/assembly/models/persistent_tokens.ts:141]: Same for the logic below this line,
You can just do `this.add(tokenId, token, accountId)` with the new info

File: as/cNFT/assembly/models/persistent_tokens.ts [hunk 4]
```diff
@@ -128,6 +172,18 @@ export class PersistentTokens {
...
+        tokenId: TokenId,
+        accountId: AccountId
+    ): void {
+        let accountTokenSet = this._amap.getSome(accountId)
+
+        if(accountTokenSet) {
+            accountTokenSet.delete(tokenId);
+            this._amap.set(accountId, accountTokenSet);
+        }
+    }
```
Comment (achrafismyname) [inline as/cNFT/assembly/models/persistent_tokens.ts:186]: Shouldn't this return `accountTokenSet`?

**Review by achrafismyname — CHANGES_REQUESTED** [review]
it looks good but I think it needs some changes first

File: as/cNFT/assembly/models/persistent_tokens.ts (hunk 2 above)
Comment (AshenI99) [reply as/cNFT/assembly/models/persistent_tokens.ts:125]: Yeah. agreed :-)

File: as/cNFT/assembly/models/persistent_tokens.ts (hunk 4 above)
Comment (AshenI99) [reply as/cNFT/assembly/models/persistent_tokens.ts:186]: I have already set the new tokens using line no 184. So no need to return them I suppose.

File: as/cNFT/assembly/models/persistent_tokens.ts (hunk 3 above)
Comment (AshenI99) [reply as/cNFT/assembly/models/persistent_tokens.ts]: Yeah. Sure :-)

File: as/cNFT/assembly/models/persistent_tokens.ts (hunk 4 above)
Comment (achrafismyname) [reply as/cNFT/assembly/models/persistent_tokens.ts:182]: Ah okay that makes sense, thanks!

File: as/cNFT/assembly/models/persistent_tokens.ts [hunk 5]
```diff
@@ -113,6 +113,46 @@ export class PersistentTokens {
...
+
+    transfer(tokenId: TokenId, bidderId: AccountId): void {
+
+        /* Getting stored token from tokenId */
+
+        const token = this.get(tokenId)
+
+        if(token){
+            const accountId = token.owner_id
+
```
Comment (achrafismyname) [inline as/cNFT/assembly/models/persistent_tokens.ts:129]: Can you make this like this below, just to make it easy to read
````ts
if(!token){
  return;
}
````

**Review by achrafismyname — APPROVED** [review]
Seems perfect :-)

**Review by devey — APPROVED** [review]
Had to restructure a little bit. Maybe give it a second look to see if something doesn't look right. Should be good to go!

**Comment by AshenI99** [comment]
> Had to restructure a little bit. @AshenI99 @achrafismyname maybe give it a second look to see if something doesn't look right. Should be good to go!

@devey Seems good for me. This structure is better than previous. It makes sense for me. Thanks :-)

**Comment by achrafismyname** [comment]
Seems good!

**Pull request merged**